File: arrays/contains_duplicate/contains_duplicate.py
""" Solution Module """
class Solution:
    """ Solution Module """
    # def run_solution(self, nums):
    #     """ Solution Method """
    #     array = []
    #     for num in nums:
    #         if num in array:
    #             return True
    #         array.append(num)
    #     return False

    # Using Hash Set and length
    def run_solution(self, nums):
        """ Solution Method """
        array = set(nums)
        if len(nums) != len(array):
            return True
        return False

File: arrays/contains_duplicate/contains_duplicate.ts
export class Solution {
    /**
     * @param {string} testInput
     * @return {any} Returns either boolean to validate the test cases or a custom result.
     */
    run_solution(testInput: number[]) {
        const set = new Set(testInput);
        return set.size !== testInput.length;
    }
}
